clippy_lints/src/single_component_use_path.rs

```diff
@@ -0,0 +1,35 @@
+use crate::utils::span_lint_and_sugg;
+use rustc_lint::{EarlyLintPass, EarlyContext};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+use syntax::ast::{Item, ItemKind};
+use if_chain::if_chain;
+use rustc_errors::Applicability;
+
+declare_clippy_lint! {
+    pub SINGLE_COMPONENT_USE_PATH,
+    style,
+    "use with single component path is redundant in 2018 edition"
```

Comment:
with `cx.sess.opts.edition` you can check if the edition is [`Edition2018`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_span/edition/enum.Edition.html) and only trigger the lint for the 2018 edition.

Reply:
I just tested it with the 2015 edition and it also works there, without extern crate and without use: [Playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2015&gist=a3422242a0dc2126371d7f96a5591247)

Reply:
@flip1995, I have read your example on playground. I saw no 'extern cratename;' item. Does playground handle that? Your discussion with @jhpratt implies external crate must be declared to use, did playground add that under the cover?

Reply:
It's not a playground feature. You can test it locally:

```toml
# Cargo.toml
[package]
name = "woec"
version = "0.1.0"
authors = ["flip1995 <hello@philkrones.com>"]
edition = "2015"

[dependencies]
bit-vec = "*"
```

```rust
// src/main.rs
fn main() {
    bit_vec::BitVec::new();
}
```

`cargo build` should run without any errors.

Reply:
It should be a common feature rather than 2018 edition specific. I will tweak the test and logic then.

Reply:
I think 'bit_vec' is a better candidate for test rather than my origin std path, I am not happy with my origin 'use std;' case, is there a toml file to add dependencies for ui tests?

Reply:
No, we can't add toml files for test deps. But you can use an existing dep, like serde or regex or similar.